src/api.ts

```diff
@@ -0,0 +1,216 @@
+type httpStatusSuccessful = 200 | 201 | 202 | 203 | 204 | 205 | 206 | 207 | 208 | 226;
+type httpStatusClientError = 400 | 401 | 402 | 403 | 404 | 405 | 406 | 407 | 408 | 409 | 410 | 411 | 412 | 412 | 413 | 414 | 415 | 416 | 417 | 418 | 421 | 422 | 423 | 424 | 425 | 426 | 428 | 429 | 431 | 451;
+type httpStatusServerError = 500 | 501 | 502 | 503 | 504 | 505 | 506 | 507 | 508 | 510 | 511;
+
+type APIResponse<T extends unknown = unknown> = {
+    httpStatusCode: httpStatusSuccessful
+    body: T,
+    headers: Record<string, string>
+}
+
+class HTTPError extends Error {
+    httpStatusCode: httpStatusClientError | httpStatusServerError;
+    body: unknown;
+    headers: Record<string, string>;
+    constructor({ name, message, code, headers, body }: { name?: string, message?: string, code: httpStatusClientError | httpStatusServerError, headers: Record<string, string>, body: unknown }) {
+        super();
+        if (name) {
+            this.name = name;
+        }
+        if (message) {
+            this.message = message;
+        }
+        this.httpStatusCode = code;
+        this.headers = headers;
+        this.body = body
+    }
+}
+
+class NetworkError extends Error {
+    name = 'Network error';
+    constructor() {
+        super();
+    }
+}
+
+class BlockedError extends Error {
+    name = 'Network error';
+    constructor() {
+        super();
+    }
+}
+
+class CancelledError extends Error {
+    name = 'Request cancelled'
+    constructor() {
+        super();
+    }
+}
+
+type FetchOptions = {
+    headers?: Record<string, string>,
+    signerServiceInfo?: string,
+    timeout?: number,
+}
+
+interface JSONArray extends Array<JSONValue> { }
```

Comment:
I think the equivalent to that is `type JSONArray = Array<JSONValue>` . Seems cleaner IMO

Reply:
Good feedback thanks

Reply:
is ok if we apply this after RFC?